pkg/client/typed/dynamic/client_pool.go

```diff
@@ -42,19 +45,21 @@ func LegacyAPIPathResolverFunc(groupVersion unversioned.GroupVersion) string {
 
 // clientPoolImpl implements Factory
 type clientPoolImpl struct {
-	lock                sync.RWMutex
-	config              *restclient.Config
-	clients             map[unversioned.GroupVersion]*Client
-	apiPathResolverFunc APIPathResolverFunc
+	lock                 sync.RWMutex
+	config               *restclient.Config
+	clients              map[unversioned.GroupVersion]*Client
+	apiPathResolverFunc  APIPathResolverFunc
+	negotiatedSerializer runtime.NegotiatedSerializer
 }
 
 // NewClientPool returns a ClientPool from the specified config
-func NewClientPool(config *restclient.Config, apiPathResolverFunc APIPathResolverFunc) ClientPool {
+func NewClientPool(config *restclient.Config, apiPathResolverFunc APIPathResolverFunc, negotiatedSerializer runtime.NegotiatedSerializer) ClientPool {
```

Comment:
Because the previous code will default the codec regardless of the config.codec, so caller may have unintentionally set the config.codec to an invalid value. If namespace controller and gc are the only caller, I can move the parameter to the config.